### PR TITLE
chore(cleanup): remove stale hacks entries

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [] 

### Description

Removes a few hacks entries for hacks no longer in use 🎉 !

### Follow-ups
- [ ] we have a couple hacks for react-native-fast-image, it seems the repo is no longer active but there are a few forks, we should probably pick the most active one and see if it has integrated these fixes or contribute them back

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- remove unused hacks entries - brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: HACKS.md [hunk 1]
```diff
@@ -28,16 +28,6 @@ There was a case where echo returns 401 when a user asks for the latest echo opt
...
-## @segment+analytics-react-native-appboy patch
-
-### When can we remove this:
-
-When we upgrade to a version of `@segment/analytics-react-native` that includes an updated kotlin version compatible with the version of kotlin we need for React Native.
-
-### Explanation/Context:
-
-When updating to rn-0.69.10 we had to patch this due to kotlin version missmatch.
-
```
Comment (brainbicycle) [inline HACKS.md:40]: now that we have upgraded to newer braze package this is not needed and patch no longer exists

File: HACKS.md [hunk 2]
```diff
@@ -50,16 +40,6 @@ https://github.com/ivpusic/react-native-image-crop-picker/pull/1354
...
-## exporting MockResolverContext (@types/relay-test-utils patch-package)
-
-#### When can we remove this:
-
-Not really needed to be removed, unless it causes problems.
-
-#### Explanation/Context:
-
-We use this type in out code for our tests and the `resolveMostRecentRelayOperation`, so we exported it.
-
```
Comment (brainbicycle) [inline HACKS.md:62]: this type is exported by default in later versions that we are on

**Review by MrSltun — APPROVED** [review]
Thank you Brian 🧹🧹🧹

File: HACKS.md [hunk 3]
```diff
@@ -68,35 +48,7 @@ Doesn't really need to be removed but can be if view hierarchy issue is fixed in
...
-react-navigation has a bug with nested independent `NavigationContainer` instances. https://github.com/react-navigation/react-navigation/issues/8611
-
-#### Explanation/Context:
-
-Our patch alleviates the issue in our case, but would not work as an upstream PR.
-
-To remove this hack we can do one of two things:
-
-- Stop using nested navigation containers.
-- Fix `@react-navigation/core` properly upstream.
```
Comment (brainbicycle) [inline HACKS.md:86]: i believe this was fixed by not using nested nav containers with the nav refactor, props @MounirDhahri

File: HACKS.md [hunk 4]
```diff
@@ -68,35 +48,7 @@ Doesn't really need to be removed but can be if view hierarchy issue is fixed in
...
-#### When can we remove this:
-
-We can remove these hacks when they don't matter anymore. Neither are likely to be fixed by facebook.
-
-#### Explanation/Context:
-
-There are two hacks here:
-
-- We hack the output of the compiler to provide clickable links for error messages. Relay assumes that you put your `__generated__` folder in the root of your project, but we put it in `src`.
-- We make sure that files which do not change are not overwritten. This prevents excessive reloading by metro.
```
Comment (brainbicycle) [inline HACKS.md:99]: been gone for a while

File: HACKS.md [hunk 5]
```diff
@@ -226,16 +178,6 @@ Once we can figure out how to mock `global.setImmediate` with `global.setTimeout
...
 
-## Providers.tsx LegacyTheme
-
-#### When can we remove this:
-
-Once we have removed the `palette` directory from eigen.
-
-#### Explanation/Context:
-
-Look at the tech plan here: https://www.notion.so/artsy/palette-mobile-in-eigen-c5e3396302734f0a921aed3978f5dbeb
```
Comment (brainbicycle) [inline HACKS.md:237]: palette directory has been gone a while

File: HACKS.md [hunk 6]
```diff
@@ -249,17 +191,6 @@ patch.
...
-## Patch-package for @braze/react-native-sdk
-
-#### When can we remove this:
-
-When we upgrade @braze/react-native-sdk to version >= 8.3.0.
-
-#### Explanation/Context:
-
-We had to patch Braze in order to proceed with the react native upgrade to 0.73.9. The patch was needed to support kotlin jvm target 17 and
-in order to also make targetCompatibility and sourceCompatibility compatible with the JAVA 17 which is the standard for newer rn versions starting 0.73.
```
Comment (brainbicycle) [inline HACKS.md:261]: braze upgrades removed need for this

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (remove unused hacks entries - brian - brainbicycle)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 49974ab44ebd6badda9414f198f8bddd31cc9d9a
</p>

**Pull request merged**